src/diff/index.js

```diff
@@ -382,7 +381,7 @@ function catchErrorInComponent(error, component) {
 						continue;
 					}
 				}
-				else if (component.constructor.getDerivedStateFromError!=null) {
```

Comment:
Wondering why getDerivedStateFromError is called on the constructor instead of directly on the component here?

Reply:
I think it's the same story as for the other static method `getDerivedStateFromProps` also called on newType which is the constructor.

Reply:
I now get why this is needed. `getDerivedStateFromError` is static and thus not available on the instance but on the constructor only 🤦‍♂